lib/subset.cc

```diff
@@ -1350,20 +1355,22 @@ void SubsetPartition::merge_from_disk(string other_filename)
             diskp = (PartitionID *) (buf + i);
             i += sizeof(PartitionID);
 
-            if (!(*diskp != 0)) {		// sanity check.
-                throw khmer_exception();
-            }
+            assert((*diskp != 0)); // sanity check!
```

Comment:
Fair enough!

On Thu, Jun 4, 2015 at 6:46 PM, C. Titus Brown <notifications@github.com>
wrote:

> In lib/subset.cc
> <https://github.com/dib-lab/khmer/pull/1048#discussion_r31782427>:
>
> > @@ -1350,20 +1355,22 @@ void SubsetPartition::merge_from_disk(string other_filename)
> >              diskp = (PartitionID *) (buf + i);
> >              i += sizeof(PartitionID);
> >
> > -            if (!(*diskp != 0)) {		// sanity check.
> > -                throw khmer_exception();
> > -            }
> > +            assert((*diskp != 0)); // sanity check!
>
> On Thu, Jun 04, 2015 at 06:38:49PM -0700, Camille Scott wrote: > @@
> -1350,20 +1355,22 @@ void SubsetPartition::merge_from_disk(string
> other_filename) > diskp = (PartitionID *) (buf + i); > i +=
> sizeof(PartitionID); > > - if (!(*diskp != 0)) { // sanity check. > - throw
> khmer_exception(); > - } > + assert((*diskp != 0)); // sanity check!
> Thought we were trying to avoid `assert`s in the C++ code?
> I was unable to think of a way to trigger this in a test, and this is a
> reasonable check on things when debugging or modifying other code. I think
> we're trying to avoid asserts as things that are actually raised for users,
> but I don't see any reason not to leave them in as sanity checks.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1048/files#r31782427>.
>



-- 
Camille Scott

Department of Computer Science
Lab for Data Intensive Biology
University of California, Davis

camille.scott.w@gmail.com